smithy-build/build.gradle.kts

```diff
@@ -21,4 +21,6 @@ extra["moduleName"] = "software.amazon.smithy.build"
 dependencies {
     api(project(":smithy-utils"))
     api(project(":smithy-model"))
+
+    implementation("com.google.code.findbugs:jsr305:3.0.2")
```

Comment:
Yeah I agree there is a complicated legacy with all of these various libraries.  Thanks for spending the time to dig into them and provide the links.  It seems that [there may be an approach](https://github.com/Kotlin/KEEP/blob/41091f1cc7045142181d8c89645059f4a15cc91a/proposals/jsr-305-custom-nullability-qualifiers.md) to utilize custom annotations to prevent taking a dependency, however it's not clear to me if JSR-305 dependency is still required or not.  In any case, if you don't see any obvious roadblocks to this approach @mtdowling , I can investigate this approach further.  LMK

Reply:
I'm definitely interested in this PR landing in some form. Would the Kotlin annotation package work in the way we want (i.e., assume non-null by default)? I'm also very interested in the possibility of not needing a dependency though I don't know if that's realistic... I guess all we'd need for the case of this PR is to ensure that Kotlin's compiler supports the feature.